eidas-saml-parser/src/main/java/uk/gov/ida/notification/eidassaml/logging/EidasAuthnRequestAttributesLogger.java

```diff
@@ -15,10 +15,11 @@ public static void logAuthnRequestAttributes(AuthnRequest authnRequest) {
             MDC.put("eidasIssueInstant", authnRequest.getIssueInstant().toString());
             MDC.put("eidasIssuer", authnRequest.getIssuer().getValue());
             log.info("Authn request validated by ESP");
-        } catch (Exception e) {
-            throw e;
         } finally {
-            MDC.clear();
```

Comment:
What was the issue with `clear()`? I've noticed it's been removed in a few places. Was it removing more than it should?

Reply:
So from my understanding is if there is any MDC requests being processed in parallel, then it would be cleared as well. Specifying remove for each MDC attribute, gives you more certainty that only those will be cleared.

Reply:
[The docs](https://logback.qos.ch/manual/mdc.html#AdvancedUse) seem to suggest that each thread has its own MDC, so shouldn't be affected. I'm fine with the way this is btw, just trying to understand.

Reply:
Hmm that looks interesting. I'll have a read of that. Cheers Chris.